packages/devtools_app/lib/src/charts/chart.dart

```diff
@@ -0,0 +1,588 @@
+// Copyright 2020 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:math';
+import 'dart:ui';
+
+import 'package:flutter/cupertino.dart';
+import 'package:flutter/material.dart';
+
+import '../auto_dispose_mixin.dart';
+import 'chart_controller.dart';
+import 'chart_trace.dart';
+
+class Chart extends StatefulWidget {
+  Chart(this.controller, String title) {
+    controller.title = title;
+  }
+
+  final ChartController controller;
+
+  @override
+  ChartState createState() => ChartState();
+}
+
+class ChartState extends State<Chart> with AutoDisposeMixin {
+  ChartController controller;
+
+  @override
+  void initState() {
+    super.initState();
+    controller = widget.controller;
+  }
+
+  @override
+  void dispose() {
+    controller.dispose();
+    super.dispose();
+  }
+
+  @override
+  void didChangeDependencies() {
```

Comment:
this should be didUpdateWidget not didUpdateDependencies. It seems like the state is changing based on the widget not some inherited widget.

Reply:
Maybe I'm confused.  This is the how I thought we hooked up first time things.  didUpdateDependencies is guaranteed to be called immediately after initState I use to to connect up listeners.